packages/eslint-plugin-turbo/docs/rules/no-uncached-env-vars.md

```diff
@@ -0,0 +1,80 @@
+# Ensure all environment variables are correctly included in cache keys (`no-uncached-env-vars`)
+
+Ensures that all usages of environment variables are correctly included in cache keys. This ensures build outputs remain correctly cacheable across environments.
+
+## Rule Details
+
+This rule aims to prevent users from forgetting to include an environment variable in their `turbo.json` configuration
+
+The following examples assume the following code:
+
+```js
+const client = MyAPI({ token: process.env.MY_API_TOKEN });
+```
+
+Examples of **incorrect** code for this rule:
+
+```json
+{
+  "pipeline": {
+    "build": {
+      "dependsOn": ["^build"],
+      "outputs": ["dist/**", ".next/**"]
+    },
+    "lint": {
+      "outputs": []
+    },
+    "dev": {
+      "cache": false
+    }
+  }
+}
+```
+
+Examples of **correct** code for this rule:
+
+```json
+{
+  "globalDependencies": ["$MY_API_TOKEN"]
+  "pipeline": {
+    "build": {
+      "dependsOn": ["^build"],
+      "outputs": ["dist/**", ".next/**"]
+    },
+    "lint": {
+      "outputs": []
+    },
+    "dev": {
+      "cache": false
+    }
+  }
+}
+```
+
+```json
+{
+  "pipeline": {
+    "build": {
+      "dependsOn": ["^build", "$MY_API_TOKEN"],
```

Comment:
Can we actually detect package vs. task vs. global?

Reply:
I did spend some time thinking about this, and I don't believe we could reliably infer this (without actually running the tasks)

Reply:
I made the comment because the `correct` patterns were in different locations and a _tiny_ bit confusing. I don't think there is anything wrong with it; but it did make me pause.